src/components/Navigation/index.js

```diff
@@ -0,0 +1,134 @@
+import React, {Component, PropTypes} from 'react'
+import {NavLink} from 'react-router-dom'
+import {map, isFunction, startsWith} from 'lodash'
+import Icon from 'components/Icon'
+import Logo from 'components/Logo'
+
+const sharedLinkClassnames = `
+  pointer
+  f6
+  pv3 pv4-ns ph4 ph3-ns
+  ttu
+  no-underline
+  white-60 
+`
+
+const activeLinkClassnames = 'bl bl-0-ns bb-ns bw2 bw1-ns b--orange white'
+
+export default class Navigation extends Component {
+  
+  static propTypes = {
+    items: PropTypes.arrayOf(PropTypes.shape({
+      text: PropTypes.string.isRequired,
+      action: PropTypes.oneOfType([
+        React.PropTypes.string,
+        React.PropTypes.func,
+      ]).isRequired,
+    })).isRequired,
```

Comment:
When a prop is an object or array I like to use a specific shape proptype like this instead of a generic `array` or `object` so users know exactly what to send.